robot-server/robot_server/service/json_api/response.py

```diff
@@ -88,7 +79,40 @@ class MultiResponse(
     links: ResponseLinksT = Field(..., description=DESCRIPTION_LINKS)
 
 
-# TODO(mc, 2021-12-09): remove this model prior to 5.0 prod release
+ResponseBodyT = TypeVar("ResponseBodyT", bound=BaseResponseBody)
+
+
+class PydanticResponse(JSONResponse, Generic[ResponseBodyT]):
+    """A custom JSON response that uses Pydantic for JSON serialization.
+
+    Pydantic's JSON serialization is much more performant than the
+    default FastAPI serialization.
+    """
```

Comment:
```suggestion
    """A custom JSON response that uses Pydantic for JSON serialization.

    Returning this class from an endpoint function is much more performant
    than returning a plain Pydantic model and letting FastAPI serialize it.
    """
```

This feels a little clearer to me for someone coming into this cold.